### ISSUE TITLE
window manipulation issues

### ISSUE THREAD

**Issue by scandum** [mintty/mintty#899]
Mintty doesn't handle a few of the \e[ ... t options correctly. I'll use row,col aka y,x for all notations.

My display resolution: 1080, 1920

Easiest issue to test is moving the terminal.

echo -e "\e[3;0;0t"

Instead of moving the terminal to position 0,0 it moves the terminal to position 0,8

When requesting the position of a maximized screen using:

echo -e "\e[13t"

Mintty returns: \e[3;-8;-8t

This should of course return \e[3;0;0t. I assume mintty justifies the screen to the right, this should probably be changed to justifying to the left to address this problem.

There is also an issue with mintty reporting the screen dimensions while maximized:

echo -e "\e[14;2t"

Mintty returns: \e[4;1066;1936t acknowledge the width is larger than 1920.

Mintty reports a correct width of 1920 while in full screen (alt+enter) mode.

Awesome terminal by the way, keep up the great work. :)

**Comment by BrianInglis** [comment]
Seeing the same type of results - screen 1920x1200 - window 120x60+210+120: 
```
$ xterm_info
window position 3 px 210+120    ^[[13t          ^[[3;210;120t$
window size     4 px 875x881    ^[[14;2t        ^[[4;881;875t$
text position   3 px 219+152    ^[[13;2t        ^[[3;219;152t$
text size       4 px 823x840    ^[[14t          ^[[4;840;823t$
text chars      8 ch 120x60     ^[[18t          ^[[8;60;120t$
char size       6 px 7x14       ^[[16t          ^[[6;14;7t$
screen chars    9 ch 274x85     ^[[19t          ^[[9;85;274t$
screen size     5 px 1920x1200  ^[[15t          ^[[5;1200;1920t$
window status   2 icon          ^[[11t          ^[[2t$
```
maximized: 
```
$ xterm_info
window position 3 px -8+-8      ^[[13t          ^[[3;-8;-8t$
window size     4 px 1936x1216  ^[[14;2t        ^[[4;1216;1936t$
text position   3 px 1+24       ^[[13;2t        ^[[3;1;24t$
text size       4 px 1884x1175  ^[[14t          ^[[4;1175;1884t$
text chars      8 ch 271x83     ^[[18t          ^[[8;83;271t$
char size       6 px 7x14       ^[[16t          ^[[6;14;7t$
screen chars    9 ch 274x85     ^[[19t          ^[[9;85;274t$
screen size     5 px 1920x1200  ^[[15t          ^[[5;1200;1920t$
window status   2 icon          ^[[11t          ^[[2t$
```
looks like e.g. twips is being rounded up to 8 pixels but is incrementing the sizes by 8 - missing a decrement before a divide and multiply - something like `n/pixel_twips/CHAR_BIT*CHAR_BIT` instead of `((n - 1)/pixel_twips - 1)/CHAR_BIT*CHAR_BIT`

**Comment by mintty** [comment]
> echo -e "\e[3;0;0t"
> Instead of moving the terminal to position 0,0 it moves the terminal to position 0,8

Works fine for me, except with option `-B frame` (default or `-B void` work).

> When requesting the position of a maximized screen using:
> echo -e "\e[13t"
> Mintty returns: \e[3;-8;-8t

Fixed the negative case (modulo ushort).
Where the offset comes from, I do not know. 
Note, however, that xterm also exhibits some weird deviations under Cygwin/X (not under Linux).

> There is also an issue with mintty reporting the screen dimensions while maximized:
> echo -e "\e[14;2t"
> Mintty returns: \e[4;1066;1936t acknowledge the width is larger than 1920.

That accounts for the border which is outside the screen in this case.

**Comment by mintty** [comment]
CSI 14 t report would include outer padding if I interpret xterm's ctlseqs.txt correctly.
Please try the impact of setting `Padding`.

About sixel display, I've just fixed a bug that would occur when you output a really large number of sixel images. Maybe that's in fact possible when dragging around an image with such a framework.
This should be a separate issue, and it will help if you describe a completely reproducible test case, cygwin-only preferred if possible.

**Comment by mintty** [comment]
If I scale down an 80x24 window to 14x7 cell size (as reported via CSI 16 t), CSI 14 t reports 336x560 for me which looks correct.
Font is DejaVu Sans Mono with an effective Leading of 0, if that matters (what's your's?), no scrollbar.
With scrollbar enabled, there is apparently some incorrect adjustment...

**Comment by mintty** [comment]
Released 3.0.3.
